tobiasHe/quiz/q1.js

```diff
@@ -0,0 +1,46 @@
+//Given an array of positive numbers and a positive number ‘k,’ find the maximum sum of any contiguous subarray of size ‘k’.
+
+//use sliding window because it's a subarray
+//the window should slide when the size of the window exeeds length of k - 1
+//subtract from the currentSum when we slide the windowStart
+//keep track of the maxSum at each iteration.
+//return maxSum
+
+const maxSum = function(arr, k) {
+	let maxSum = 0;
+	let currSum = 0;
+	let windowStart = 0;
+
+	if (k < 1) {
+		return maxSum;
+	} else {
+		for (let windowEnd = 0; windowEnd < arr.length; windowEnd++) {
+			currSum += arr[windowEnd];
+			if (windowEnd >= k - 1) {
+				maxSum = Math.max(currSum, maxSum);
+				currSum -= arr[windowStart];
+				windowStart++;
+			}
+		}
+	}
+	return maxSum;
+};
+
+//test cases
+let arr = [];
+let arr2 = [ 1, 2, 3, 4, 5 ];
```

Comment:
Would recommend more test cases, especially for an unsorted array because you want to make sure your solution works for both an unsorted and sorted array.